### PR TITLE
Allow null args to lists/maps/setsEqual

### PR DESCRIPTION
**Pull request by cbracken**
This updates the parameters to listsEqual, mapsEqual and setsEqual in
the collection utils library to nullable types.

This avoids forcing callers to manually check for null, is a bit more in
line with Object.equals() which handles the null check in the runtime,
and avoids a breaking change to users who have not yet transitioned to
non-null by default.

Related: #606.

### CODE REVIEW

File: lib/src/collection/utils.dart [hunk 1]
```diff
@@ -49,8 +51,9 @@ bool mapsEqual(Map a, Map b) {
 ///
 /// Returns `true` if [a] and [b] are both null, or they are the same length and
 /// every element in [b] exists in [a].
-bool setsEqual(Set a, Set b) {
+bool setsEqual(Set? a, Set? b) {
   if (a == b) return true;
+  if (a == null || b == null) return false;
```
Comment (yjbanov) [inline lib/src/collection/utils.dart:56]: Do we need to test the null cases?

**Review by yjbanov — APPROVED** [review]
lgtm

File: lib/src/collection/utils.dart (hunk 1 above)
Comment (cbracken) [reply lib/src/collection/utils.dart:56]: Yep good call -- done.
